mne/channels/channels.py

```diff
@@ -987,6 +989,118 @@ def add_channels(self, add_list, force_update_info=False):
             assert all(len(r) == self.info['nchan'] for r in self._read_picks)
         return self
 
+    def combine_channels(self, group_by, method='mean'):
+        """Combine channels based on regions of interest (ROIs).
+
+        Parameters
+        ----------
+        group_by : dict
+            Specifies which channels are aggregated into a single channel, with
+            aggregation method determined by the ``method`` parameter. One new
+            pseudo-channel is made per dict entry; the dict values must be
+            lists of picks (integer indices of ``ch_names``). For example::
+
+                group_by=dict(Left_ROI=[1, 2, 3, 4], Right_ROI=[5, 6, 7, 8])
+
+            Note that within a dict entry all channels must have the same type.
+        %(method)s
+            If callable, the callable must accept one positional input (data of
+            shape ``(n_channels, n_times)``, ``(n_epochs, n_channels,
+            n_times)``, or ``(n_evokeds, n_channels, n_time)``) and return an
+            :class:`array <numpy.ndarray>` of shape ``(n_times)``, ``(n_epochs,
+            n_times)``, or ``(n_evokeds, n_times)``. For example with an
+            instance of Raw::
+
+                method = lambda data: np.median(data, axis=0)
+
+            Defaults to ``mean``.
+
+        Returns
+        -------
+        inst : instance of Raw, Epochs, or Evoked
+            A new instance.
+        """
+        from ..io import BaseRaw, RawArray
+        from ..epochs import BaseEpochs
+        from ..evoked import Evoked
+        from .. import EpochsArray, EvokedArray
+
+        # Check to see if data is preloaded
+        _check_preload(self, 'Combining channels')
+
+        ch_axis = len(self._data.shape) - 2  # idx of the channel axis
+        ch_idx = _picks_to_idx(self.info, None)
+        ch_types = _get_channel_types(self.info)
+        group_by = deepcopy(group_by)
+
+        # Convert string values of ``method`` into callables
+        if isinstance(method, str):
+            method_dict = {key: partial(getattr(np, key), axis=ch_axis)
+                           for key in ('mean', 'median', 'std')}
+            try:
+                method = method_dict[method]
+            except KeyError:
+                raise ValueError('"method" must be a callable, or one of '
+                                 '"mean", "median", or "std"; got {}'
+                                 ''.format(method))
```

Comment:
is there a way to modify the `_make_combine_callable` function to be usable here (rather than copy-pasting parts of it)?

https://github.com/mne-tools/mne-python/blob/056c273f1bb96bb3229e60874122f830f946746c/mne/viz/utils.py#L2975-L2998

Reply:
This function was pretty handy but not exactly what I wanted, as `combine_channels` does not accept None or 'gfp' currently. In this case I feel like it would probably make `_make_combine_callable` messy to modify it (in terms of documentation with the docdict, and error messages). Please let me know what you think.
*&rarr; Linked to https://github.com/mne-tools/mne-python/pull/8014#discussion_r455740903*